File: week-1/problems/mario-more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // TODO: prompt for pyramid height between 1 & 8 inclusive & rpt. until valid
    int height;
    do
    {
        height = get_int("Enter the pyramid height plz: ");
    }
    while (height < 1 || height > 8);

    // TODO: print 1 row for each height
    for (int i = 0; i < height; i++)
    {
        // TODO: print 1 col for each height, with space & (i + 1) for hashes
        for (int j = 0; j < (height + 2 + i + 1); j++)
        {
            // Middle spaces == height & height + 1
            if (j == height || j == height + 1)
            {
                printf(" ");
            }
            // Leading spaces
            else if (j < height - i - 1)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

File: week-6/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    # Get input
    height = get_height()
    # Print blocks
    print_blocks(height)


def get_height():
    # Get height between 1 & 8 inclusive
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    return height


def print_blocks(height):
    # Print 2 inverse pyramids with 2 spaces between each
    for i in range(height):
        for j in range(height + i + 3):
            if (j < height - i - 1 or j > height + i + 2
                    or (j > height - 1 and j < height + 2)):
                print(" ", end="")
            else:
                print("#", end="")
        print()


if __name__ == "__main__":
    main()
